acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/keymanager/ImportKeystoresAcceptanceTest.java

```diff
@@ -0,0 +1,137 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.tests.keymanager;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+
+import tech.pegasys.web3signer.core.service.http.handlers.keymanager.imports.ImportKeystoresRequestBody;
+import tech.pegasys.web3signer.core.signing.KeyType;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.tuweni.bytes.Bytes;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.jupiter.api.Test;
+
+public class ImportKeystoresAcceptanceTest extends KeyManagerTestBase {
+
+  private static final String PUBLIC_KEY =
+      "0x98d083489b3b06b8740da2dfec5cc3c01b2086363fe023a9d7dc1f907633b1ff11f7b99b19e0533e969862270061d884";
+
+  @Test
+  public void invalidRequestBodyReturnsError() {
+    setupSignerWithKeyManagerApi();
+    final Response response = callImportKeystores("{\"invalid\": \"json body\"}");
+    response.then().assertThat().statusCode(400);
+  }
+
+  @Test
+  public void validRequestBodyReturnsSuccess() throws IOException, URISyntaxException {
+    setupSignerWithKeyManagerApi();
+    final Response response = callImportKeystores(composeRequestBody());
+    response
+        .then()
+        .contentType(ContentType.JSON)
+        .assertThat()
+        .statusCode(200)
+        .body("data.status", hasItem("imported"));
+  }
+
+  @Test
+  public void existingKeyReturnsDuplicate() throws IOException, URISyntaxException {
+    createBlsKey("eth2/bls_keystore.json", "somepassword");
+    setupSignerWithKeyManagerApi();
+
+    assertThat(signer.listPublicKeys(KeyType.BLS).size()).isEqualTo(1);
+
+    final Response response = callImportKeystores(composeRequestBody());
+    response
+        .then()
+        .contentType(ContentType.JSON)
+        .assertThat()
+        .statusCode(200)
+        .body("data.status", hasItem("duplicate"));
+
+    assertThat(signer.listPublicKeys(KeyType.BLS).size()).isEqualTo(1);
+  }
+
+  @Test
+  public void importLoadsNewKeys() throws IOException, URISyntaxException {
+    setupSignerWithKeyManagerApi();
+    assertThat(signer.listPublicKeys(KeyType.BLS).size()).isEqualTo(0);
+
+    callImportKeystores(composeRequestBody()).then().statusCode(200);
+
+    validateApiResponse(callListKeys(), "data.validating_pubkey", hasItem(PUBLIC_KEY));
+    assertThat(signer.listPublicKeys(KeyType.BLS).size()).isEqualTo(1);
+    assertThat(signer.listPublicKeys(KeyType.BLS).get(0)).isEqualTo(PUBLIC_KEY);
```

Comment:
Also important to check that the status is "imported"

Reply:
this checks that the list endpoint gets the new pubkey, but there's no status on the list response

Reply:
Whoops added this wrong line